### ISSUE TITLE
Flex Question + non-dev environment

### ISSUE THREAD

**Issue by inverse** [symfony/maker-bundle#151]
How should this work with flex and non-dev environments? E.g. it specifies this to be included as a dev composer dependency and adds the following to my `bundles.php`:
```
    Symfony\Bundle\MakerBundle\MakerBundle::class => ['dev' => true],
```

When deploying code to production It's often advised to run `--no-dev` yet this will cause the application to break.

Any advice?

**Comment by inverse** [comment]
@weaverryan thanks for getting back to me on this.

Totally overlooked this one - I was missing the injection of the environment variables in CLI for doing the composer install step

**Comment by Gemorroj** [comment]
@weaverryan but in `config/bundles.php` missing classes are specified.
![-1](https://user-images.githubusercontent.com/885731/39363274-60023cf0-4a32-11e8-8dec-f6fe4e0c06e7.png)

**Comment by inverse** [comment]
@Gemorroj you need to set the `APP_ENV` environment variable to `prod` when running composer with `--no-dev`

**Comment by dsuurlant** [comment]
You're right, we were running into a different issue with the environment. Thanks!

**Comment by didacrios** [comment]
Hi all,

Seems I'm running similar problems, I tried to follow @weaverryan  approach but with no luck. I guess I'm missing something

I am just trying to run some commands to deploy our symfony application following this commands:

```bash
composer dump-env prod
composer install -o --no-dev
php bin/console make:migration
php bin/console doctrine:migrations:migrate
```

My  `symfony/maker-bundle` contains the following version `^1.13` also I checked the `.env.local.php` generated in the server and the `APP_ENV` is set to `prod`

With all of this, when running the `php bin/console make:migration` it throws the folowing error:

There are no commands defined in the "make" namespace.      

![image](https://user-images.githubusercontent.com/211009/66025723-6f7c8f80-e4f7-11e9-9a46-4d84a8e7eaac.png)

Any idea of what we are missing?

**Comment by maxhelias** [comment]
Why you running the command `php bin/console make:migration` for production install ?

**Comment by didacrios** [comment]
Shouldn't we?

**Comment by maxhelias** [comment]
Indeed, this is to create migrations from the difference of your model to your base.
It is very risky to create a migration in prod and then run it without seeing the result.
Remove the command and it will be good 😃

**Comment by didacrios** [comment]
So your recommendation is that this commands should be run manually?

```php bin/console make:migration
php bin/console doctrine:migrations:migrate
```

**Comment by maxhelias** [comment]
Only the first and in dev mode, keep the second for production.

**Comment by didacrios** [comment]
Thanks @maxhelias :)
